src/Microsoft.AspNetCore.Mvc.Core/Authorization/AuthorizeFilter.cs

```diff
@@ -46,6 +77,12 @@ public AuthorizeFilter(AuthorizationPolicy policy)
                 throw new ArgumentNullException(nameof(context));
             }
 
+            Policy = Policy ?? await AuthorizationPolicy.CombineAsync(PolicyProvider, AuthorizeData);
+            if (Policy == null)
+            {
+                return;
```

Comment:
Should we cache the fact that `Combine` was called? The current code will always call `Combine` in the event it returns null and a Policy isn't set.